paper-button.html

```diff
@@ -44,154 +44,131 @@
+    is: 'paper-button',
+
+    properties: {
+
+      /**
+       * If true, the user cannot interact with this button.
+       *
+       * @attribute disabled
+       * @type boolean
+       * @default false
+       */
+      disabled: {
+        type: Boolean
+      },
 
-    <core-a11y-keys keys="space enter" target="{{}}" on-keys-pressed="{{_activate}}"></core-a11y-keys>
+      /**
+       * If true, the button is currently active either because the
+       * user is touching it, or the button is a toggle
+       * and is currently in the active state.
+       *
+       * @attribute active
+       * @type boolean
+       * @default false
+       */
+      active: {
+        type: Boolean
+      },
 
-  </template>
+      /**
+       * If true, the button currently has focus.
+       *
+       * @attribute focused
+       * @type boolean
+       * @default false
+       */
+      focused: {
+        type: Boolean
+      },
 
-  <script>
-    Polymer({
-
-      publish: {
-
-        /**
-         * If true, the button will be styled with a shadow.
-         *
-         * @attribute raised
-         * @type boolean
-         * @default false
-         */
-        raised: false,
-
-        /**
-         * By default the ripple emanates from where the user touched the button.
-         * Set this to true to always center the ripple.
-         *
-         * @attribute recenteringTouch
-         * @type boolean
-         * @default false
-         */
-        recenteringTouch: false,
-
-        /**
-         * By default the ripple expands to fill the button. Set this to true to
-         * constrain the ripple to a circle within the button.
-         *
-         * @attribute fill
-         * @type boolean
-         * @default true
-         */
-        fill: true
+      /**
+       * If true, the button toggles the active state with each tap.
+       * Otherwise, the button becomes active when the user is holding
+       * it down.
+       *
+       * @attribute toggle
+       * @type boolean
+       * @default false
+       */
+      toggle: {
+        type: Boolean
+      },
 
+      /**
+       * If true, the button should be styled with a shadow.
+       *
+       * @attribute raised
+       * @type boolean
+       * @default false
+       */
+      raised: {
+        type: Boolean
       },
 
-      _activate: function() {
-        this.click();
-        this.fire('tap');
-        if (!this.pressed) {
-          var bcr = this.getBoundingClientRect();
-          var x = bcr.left + (bcr.width / 2);
-          var y = bcr.top + (bcr.height / 2);
-          this.downAction({x: x, y: y});
-          var fn = function fn() {
-            this.upAction();
-            this.removeEventListener('keyup', fn);
-          }.bind(this);
-          this.addEventListener('keyup', fn);
-        }
+      /**
+       * The elevation of the button, represented as 2dp. Note that this
+       * is a computed property and should be treated as read-only.
```

Comment:
While it is true that this is `readOnly` in this case, this is a good representative sample of how the composed children "embue" their host with behavior more generally.

I think it would be cool for it to be supported, but even if it was, it doesn't solve the performance problem.